src/benchmarks/micro/runtime/BenchmarksGame/binarytrees-6.cs

```diff
@@ -0,0 +1,123 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+// Adapted from binary-trees C# .NET Core #6 program
+// https://salsa.debian.org/benchmarksgame-team/benchmarksgame/-/blob/master/public/download/benchmarksgame-sourcecode.zip
+// Best-scoring C# .NET Core version as of 2020-08-12
+
+// The Computer Language Benchmarks Game
+// https://salsa.debian.org/benchmarksgame-team/benchmarksgame/
+//
+// contributed by Marek Safar
+// concurrency added by Peperud
+// fixed long-lived tree by Anthony Lloyd
+// ported from F# version by Anthony Lloyd
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+using BenchmarkDotNet.Attributes;
+using MicroBenchmarks;
+
+namespace BenchmarksGame
+{
+    [MaxIterationCount(40)] // the default 20 is not enough, the benchmark has multimodal distribution and needs more runs
+    [BenchmarkCategory(Categories.Runtime, Categories.BenchmarksGame)]
```

Comment:
recently we have introduced `JIT` category and added all `BenchmarksGame` benchmarks to it: https://github.com/dotnet/performance/pull/1437

 it would be nice to add `JIT` category to this benchmark as well
```suggestion
    [BenchmarkCategory(Categories.Runtime, Categories.BenchmarksGame, Categories.JIT)]
```